framework/src/main/java/org/checkerframework/framework/util/JavaExpressionParseUtil.java

```diff
@@ -119,22 +119,31 @@
     private static final int PARAMETER_REPLACEMENT_LENGTH = PARAMETER_REPLACEMENT.length();
 
     /**
-     * Parse a string and return its representation as a {@link JavaExpression}, or throw a {@link
+     * Parse a string and view-point adapt it to the given {@code context} and {@code localPath}.
+     * Return its representation as a {@link JavaExpression}, or throw a {@link
      * JavaExpressionParseException}.
      *
+     * <p>A {@link TreePath}, {@code somePath}, is required to use the underlying javac API to
```

Comment:
What parts of `somePath` are used?  Do paths A and B always have the identical effect, so long as A and B are in the same compilation unit?  Does this path need to be in the same compilation unit as `localPath`?  If the compilation unit doesn't matter, then does this need to be passed in at all?  Could it be set as a field in `JavaExpressionParseUtil`?

I don't find `somePath` evocative.  If the compilation unit is what matters, maybe the name could reflect that, such as `pathWithinCompilationUnit`.  (I admit that is long and clunky; maybe it can be further improved.)

Reply:
My hypothesis is that when JavaExpressionParseUtil uses `somePath` rather than `localPath`, passing any path should have identical effect.  To test this, I passed `SourceChecker#currentRoot` when `useLocalScope` is false.  When I run the tests, the only errors are related to https://github.com/typetools/checker-framework/issues/862.  So, I think if I fix that bug by forcing class names to be fully-qualified, then `somePath` can just be `SourceChecker#currentRoot`.  (I'm working on that fix now.)

Reply:
After #4266, is merged, `somePath` can be `SourceChecker#currentRoot`.  The last commit makes this change.